File: cpp_brains/connection.cpp
#include "connection.h"
#include "neuron.h"

#include <cstdlib>

Connection::Connection(Neuron *inputNeuron)
    :m_inputNeuron{inputNeuron}
{
    m_weight = 1.0 * std::rand() / RAND_MAX;
}

double Connection::activate() const
{
    return m_inputNeuron->activate() * m_weight;
}

double Connection::weight() const
{
    return m_weight;
}

void Connection::setWeight(double weight)
{
    m_weight = weight;
}

void Connection::adjustWeight(double adjustment)
{
    m_weight += adjustment;
}

const Neuron &Connection::getNeuron() const
{
    return *m_inputNeuron;
}

File: kotlin_brians/src/main/kotlin/neurals/Connection.kt
package neurals

class Connection(val inputNeuron: Neuron) {
    var weight: Double = (Math.random() - 0.5) * 2.0

    fun activate(): Double {
        return inputNeuron.activate() * weight
    }

    fun adjustWeight(adjustment: Double) {
        weight += adjustment
    }
}